sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Triggers/EventHubAsyncCollector.cs

```diff
@@ -12,21 +13,21 @@
 namespace Microsoft.Azure.WebJobs.EventHubs
 {
     /// <summary>
-    /// Core object to send events to EventHub.
-    /// Any user parameter that sends EventHub events will eventually get bound to this object.
-    /// This will queue events and send in batches, also keeping under the 1024kb event hub limit per batch.
+    /// This collector allows events to be published to Event Hubs asynchronously in the background.
     /// </summary>
-    internal class EventHubAsyncCollector : IAsyncCollector<EventData>, IDisposable
+    public class EventHubAsyncCollector : IAsyncCollector<EventData>, IDisposable
```

Comment:
An alternative would be to define a new model type that contains an EventData and PartitionKey property. I think this is probably cleaner, but we may want to run it by the Functions folks as I believe this is a unique pattern we are establishing here.

Reply:
Well either way we are introducing a new type that users will need to somehow discover, so docs will be important here. But I agree that your approach is probably better for users.

Reply:
I don't see a reason to prefer the interface over this, so I converted all samples over so that we are demonstrating a unified approach rather than positioning as "just in this specific scenario"

Reply:
Sounds good - I think we will also need to update these docs - https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-event-hubs-output?tabs=in-process%2Cfunctionsv2%2Cextensionv5&pivots=programming-language-csharp#example

Reply:
I'm going to create an issue to follow-up on this one and get these docs updated once we release.

Reply:
Tracking via #33920.